neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt

```diff
@@ -35,6 +35,11 @@ class VariantSharedPreferencesRepository(val klass: Class<out Any>,
         sharedPreferences = context.getSharedPreferences("$formattedPackageName$SHARED_PREFERENCES_FILE_NAME", Context.MODE_PRIVATE)
         editor = sharedPreferences.edit()
 
+        if(!baseVariants.containsKey(defaultVariant)) {
+            editor.clear();
+            throw RuntimeException("You must declare a valid default variant");
```

Comment:
I dont like the idea of having to wrap initialise in a try/catch block. I could just log an error and return, but this kind of error is not recoverable (unless we go back to us picking a default, which has other negative implications)